### PR TITLE
Adding digital location attribute and route to view only digital locations - closes #46

### LINKED ISSUE #46: Filter delivery locations to digital contact locations
`pickup_location?` gets us really close:

```ruby
irb(main):003:0> Locations::DeliveryLocation.select { |d| d.pickup_location? }.each { |l| puts l.label } 
Plasma Physics Library
Architecture Library
East Asian Library
Engineering Library
Marquand Library of Art and Archaeology
Mendel Music Library
Stokes Library
Lewis Library, Geosciences and Map Library
Lewis Library
Firestone Library
ReCAP Reading Room
```

Only "Lewis Library, Geosciences and Map Library" and "ReCAP Reading Rooom" are incorrect; so a migration would be easy. Propose adding a boolean `digital_location` field to this model, and then adding a new route,  `/locations/delivery_locations{.json}` that uses `digital_location?` as a filter, but otherwise lists the same data as is available for DeliveryLocations.

### PR DESCRIPTION
**Pull request by escowles**

### CODE REVIEW

File: db/migrate/20160127140802_add_digital_location_to_delivery_location.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDigitalLocationToDeliveryLocation < ActiveRecord::Migration
+  def change
+    add_column :locations_delivery_locations, :digital_location, :boolean
+  end
```
Comment (jpstroop) [inline db/migrate/20160127140802_add_digital_location_to_delivery_location.rb]: Should we have the migration set true or false for existing locations, or would be better/easier to do this in the console later?

Comment (escowles) [inline db/migrate/20160127140802_add_digital_location_to_delivery_location.rb]: Setting a default seems like a good idea, to minimize the manual editing required -- should be it be true?

Comment (jpstroop) [inline db/migrate/20160127140802_add_digital_location_to_delivery_location.rb]: If we just want a default, true is fine and then manually editing to get things into a correct state.

As I said in the original issue,

```ruby
Locations::DeliveryLocation.select { |d| d.pickup_location? }
```

gets us really close; only two ("Lewis Library, Geosciences and Map Library" and "ReCAP Reading Rooom") would be incorrect; so if getting more fancy seems appropriate, that might be an idea.

Comment (escowles) [inline db/migrate/20160127140802_add_digital_location_to_delivery_location.rb]: I don't know about setting a default value that's another column.  So I've added a second migration that updates everything in the db.

**Comment by jpstroop** [comment]
:+1: wanna squash?

**Pull request merged**